---
layout: post
title: Third face to face project meeting
author: gyzhang
modified:
excerpt: A report of our third face-to-face project meeting in Gainesville from Mar 21-23, 2018.
tags: ["development plan", "project plan", "project meeting"]
published: true
comments: true
image:
  feature:
  teaser:
  thumb:
---

The [Phyloreferencing project] team had their third face to face project meeting from Mar 21-23, 2018, at the UF campus in Gainesville. A day longer than previous ones, this meeting was more extensive, but we felt the extra day was necessary to accomplish our meeting goals. We had a rather long list of items on the [agenda], and it was quite a miracle that we covered nearly all them. This is a report on the four major parts that we spent our time on at the meeting: the ontology of phyloreferences, the phyloreference curation tool (see a [prototype here]), use case development, and project organization. Overall, this meeting set a clear direction for the way ahead, laid out goals, and planned tasks for the next six months.

## Ontology of phyloreferences
We will prioritize our time till up to August to work on a project to transcribe the content of the #Phylonyms: a Companion to the PhyloCode#, essentially a list of clade definitions, that is to be published as a book in the near future, into an ontology of phyloreferences. We plan to write a paper that would describe the ontology itself, how it works, and some illustrative examples as to what research questions can be answered with it. This paper will, however, not focus on the theory of phyloreferencing and its relevance in the broad biology research context. The intent is to publish this paper more or less simultaneously with the publication of the clade definition book. We discussed the outline of this paper, which would include a brief description of phyloreferencing, a list of phyloreferences in [OWL (Web Ontology Language)] format, ontology specifications, and two use cases that will illustrate the use of the ontology of phyloreferences.

To complete these tasks that are essential for writing this paper, we will need to first get an idea of the time requirement of curating a clade definition, converting it to a phyloreference (in OWL format), and validating the phyloreference against the reference phylogeny based on which the clade definition is made. Then we will be able to assess if it is realistic to shoot for Aug/Sep as a target time to finish the paper. To demonstrate the usefulness of this ontology of phyloreferences, we will need to show that it affords capabilities not offered by the Phyloregnum database alone.

We came out with two meaningfully and potentially implementable use cases. The first is to determine relationships between phyloreferences on the original (reference) phylogeny. We ask the question "Can relationships between clade definitions be computationally determined within a given phylogeny?" We will supply a list of clade definitions and a phylogeny to the machine, and we expect it to return a list of statements about the relationships between these definitions. For example, on the phylogeny in the figure below (redrawn from [Fisher et al., 2006]), the computer should tell us that *Exostratum* and *Leucophanes* are disjoint and *Albifolium* includes *Exostratum* and *Leucophanes*. In other words, the computer should be able to determine that *Exostratum* and *Leucophanes* do not share descendants, but all the descendants of *Exostratum* and *Leucophanes* are also descendants of *Albifolium*. We would like to do this with an OWL reasoner. The same test can also be performed on a different phylogeny to determine if the relationship between the clades can be recovered.

![Part of a phylogeny extracted from Fisher et al.]({{ site.url }}/images/2018-03-23-f2fmeeting/Fisher_phylogeny.png "Part of a phylogeny showing three clades identified by resolving phylogenetic clade definitions on the phylogeny")

<figure>
  <img src="{{site.url}}//images/2018-03-23-f2fmeeting/Fisher_phylogeny.png" alt="Partial phylogeny of Calymperaceae in Fisher et al. 2006"/>
  <figcaption>Partial Partial phylogeny of Calymperaceae, on which three phylogenetic clade definitions are resolved. In this phylogeny, Exostratum and Leucophanes are disjoint. Albifolium includes both Exostratum and Leucophanes. Redrawn from Figure 10 in Fisher et al. 2006. </figcaption>
</figure>

The second use case of the ontology of phyloreferences (OPhyR?) is answer data-driven questions using the Open Tree of Life as a comprehensive base phylogeny. We came out with a list of questions as follows. (1) How many clades maintain their original relationships with the OToL? (2) How many clade definitions resolve to a clade on the OToL? (3) How many definitions resolve to a labeled node versus nodes without a label (name) on the OToL? (4) How many of the resolved clades are congruent with the intended clades (i.e., a clade that is anchored by specifiers on the original reference phylogeny). We felt that this kind of "big data" questions will be most useful to demonstrate the power of phyloreferencing and make a compelling case for the necessity of making clade definitions computable.

For this OPhyR project, we anticipated some practical challenges ahead of us. A larger percentage of the phylogenies won't be available in a machine-readable format (e.g., a Newick/Nexus file). Some of these phylogenies may not even contain all the specifiers used in a definition. Those problems will need to be resolved in an efficient way. We discussed if we need to and how to maintain this ontology of phyloreferences, e.g., how to incorporate new clade definitions as they are made and published. We decided that sustainability is outside the scope of this paper.

## Phyloreference curation tool

The Curation Tool has been the primary target of software development in the last few months. This tool will be used to produce Phyloreference Exchange (PHYX) files, which are [JSON-LD] files that describe phyloreferences and include annotated phylogenies against which they can be tested. These files can then be incorporated into the [ontology of phyloreferences], where they are currently continuously tested. Within the next few months, we would like to [add software support] for concatenating these PHYX files into a shareable ontology of phyloreferences.

Our immediate software development goal is to provide project members with an interface they can use to curate phyloreferences into PHYX files. We walked through the [latest development version] of the Curation Tool, and identified software development tasks that needed to be carried out, grouped into Github projects as follows:

1. *[Curation Tool v0.1]*: the [minimum viable product] for the Curation Tool, capable of creating, visualizing, editing and exporting PHYX files. We determined that this is largely complete, except for one important new feature -- support for renaming nodes on a phylogeny -- and additional testing to ensure that this tool is usable. We plan to reach this goal as soon as possible, ideally in the next few weeks.

2. *[Estimate rate of phyloreference curation]*: as described above, our immediate priority is to provide some estimate of the rate at which phyloreferences can be curated from RegNum. As soon as Curation Tool v0.1 is complete, it can be used to curate PHYX files and a basic rate can be measured. We plan to reach this goal as soon as possible, ideally in the next few weeks.

3. *[Curation Tool v0.2]*: implement basic reasoning using a Java-based server backend. This will likely be extremely slow from the user's perspective to begin with, but will allow PHYX files to be tested before they are incorporated into the ontology of phyloreferences. We plan to reach this goal in the next month or two, depending on how quickly we can meet the two previous goals.

4. *[Curation Tool v1.0]*: this project collects issues that will improve the usability of the software, but are not essential to any of our goals above. We can work on these after we have successfully accomplished the previous goals. Once non-developers have used the Curation Tool, their feedback can also be used to validate and prioritize these issues. This is a long term goal, and is relatively low priority when compared to other aspects of our project.

## Use case development
We went through an exercise to develop a use case of phyloreferencing. First of all, we clarified what is meant by "developing use cases" and how to do it. We realigned our perspectives and came to a consensus that a use case demonstrates describes a research problem that requires software or infrastructure development to answer it. Use cases are supposed to drive software development. We can’t prioritize use cases until we have them in front of us, so ideally we have a lot more than we can deal with. Large and complex use cases can start with a futuristic “user story” that takes the form of a narrative – it does not have to be formally defined, which allows us to think through it as a group before we formalize it into a use case. There should be some evidence that the use case is broadly applicable, and not only relevant or interesting to a few people. For instance, the evidence may come from a publication where somebody poses a question that needs a computational solution. A use case is not abstract: you create a real person in the narrative that wants to do something real, such as a systematist or an evolutionary biology, and you pretend that the necessary infrastructure is already there. The next step from use cases is mock-ups or feasibility studies. How do we develop a use case? A use case has prerequisites, pre-conditions, an objective, and post-conditions (how do you know that you’ve met the objective). Below we will describe the use case we developed during the meeting as a group.

Use case: A biologist would like to determine the relationship between a taxon according to their understanding and a clade definition on a particular phylogeny, but the person would like do it using a computer. The software should determine, minimally, congruence or incongruence between the biologist's concept of a taxon and the clade as pointed by a clade definition. What is congruence and how does  computer assess that? The taxon in their understanding can be reduced to a set of statements that can be individually tested against the phylogeny, e.g., “clade contains X”, “X is in the phylogeny but not contained in clade”, or “phylogeny does not contain X”. For this use case, we identified three prerequisites: a phylogeny, a clade definition, and a set of statements that describe the biologist’s understanding of a taxon (which also need to be computationally amenable). The following are the pre-conditions: (1) the clade definition should be resolvable on the phylogeny, (2) the set of statements can be matched to the phylogeny (“taxon contains X”, “taxon does not contain X”), and (3) if there is a statement that cannot be matched, then that implies that X was not sampled in this phylogeny. The post-conditions of this use case are two-fold. For each statement, we can correctly state that the statement is true, false or cannot be determined using this phylogeny. Besides, we show the results for each statement to the user. We thought this use case is specific enough to drive software development, but broadly relevant so that it should be interesting to biologists across the board.

## Clarifying concepts
The first critial concept is congruence. Gaurav brought up an interesting question: on a phylogeny that only samples *Homo sapiens* as a representative of Mammalia, can we say *Homo sapiens* is congruent with Mammalia on that phylogeny? To assess that statement, we will need to assess the following set of sub-statements. *Homo sapiens* is a member of of Mammalia. Mammalia is monophyletic. All other members of Mammalia as specified by a checklist are more closely related to *Homo sapiens* than to any other samples on that phylogeny. We did not quite finish this particular discussion as we carried away with clarifying some other issues around traditional versus phylogeny-based monograph. We also talked about assessing congruence when taxon sampling differs between phylogenies. If two phylogenies each only contain two of the four major lineages of a taxon, say, Campanulaceae, we would not be able to determine the relationship of these two phylogenies (or the clades recovered as Campanulaceae on respective phylogenies). The second concept is "unambiguous". What do we mean when we say a phylogenetic clade definition is unambiguous? We asserted that phylogenetic clade definitions are unambiguous and can be resolved on a phylogeny unambigously. But we did discuss a scenario where a stem-based clade definition will not resolve on a phylogeny. If on a phylogeny (not the reference phylogeny) two internal specifiers are paraphyletic with respect to the external specifier, that clade definition will not resolve to a node. We discussed the relevance of "clade content". Under phyloreferencing, the clade content is not relevant. This premise will have some major implications as to how phyloreferencing will be used and what kinds of questions that it can be used to answer. We did not get to talk about these implications.

## Project organization
In terms of project organization, we discussed mainly three issues. We planed for the upcoming Advisory Board meeting on Apr 23. We will provide a series of presentations taking up a total of 70 minutes, given by the project PIs as well as the two postdocs, to get the advisors up to speed with what we are doing. We will give an overview of the project, briefly describe the concept and method of phyloreferencing, present the curation tool and describe user stories. The second thing we discussed is our next face to face meeting. We planned to hold it in Oct in Duke, and it will be with held in conjunction the two postdocs' visit to Duke. Lastly, we briefly talked about two upcoming meetings, namely, TDWG 2018 and idigbio's Berkeley meeting. The entire team will be attending TDWG, but the attendance of idigbio's meeting will be contingent upon funding being obtained from the museum. Other organizational issues we discussed concerned how the participants of this project interact and the scope of the project.

## Pending issues (aka the questions that keep Hilmar up at night)
The two questions that really keep Hilmar up at night are (a) How are going to solve specifier matching? and (b) What are we going to produce that draws people in? We are going to be faced with a pervasive issue of missing specifiers, and that need to dealt with computationally, rather than relying on expert opinion to place a missing specifier. To determine the position of a missing specifier, we need a huge knowledge base of phylogenetic relationships between labels, which would allow us to answer the question: can we or can we not replace a node. We might have to do a whole lot of recursive querying. One immediate thought is to use the [Open Tree of Life] to query for the position of specifiers. We will need to test this. As for the second question that keeps Hilmar up at night, i.e, what draws people in, we compared to a [recent manuscript] written by Nico Franz, where he aligned clade concepts in two bird phylogenies using the approach of concept taxonomy. We were hoping to develop something of that kind to engage people, but we will be faced with a shortage of clade definitions for birds. To further explore this second issue, we talked about who are our stakeholders. We thought they can broadly be classified as those who use the phylogenetic nomenclature, and those who don't. We may need to target them differently. There is some common goal between them, which is placing a name on a phylogeny. How can we demonstrate that phyloreferencing is a far more superior approach than what is being currently done in the field of systematics? That is the question we will need to keep pondering on.

To summarize, this was an important and productive meeting, which allowed us to develop a concrete plan for the next 6 months, align perspectives and better manage project down the road.

[agenda]: https://github.com/phyloref/organization/wiki/Third-Face-to-Face-Meeting
[Phyloreferencing project]: https://www.nsf.gov/awardsearch/showAward?AWD_ID=1458604
[prototype here]: http://www.phyloref.org/curation-tool/
[Open Tree of Life]: http://www.opentreeoflife.org
[recent manuscript]: https://doi.org/10.1101/233973
[JSON-LD]: https://en.wikipedia.org/wiki/JSON-LD
[ontology of phyloreferences]: https://github.com/phyloref/curation-workflow/
[add software support]: https://github.com/phyloref/curation-workflow/projects/3
[latest development version]: http://www.ggvaidya.com/curation-tool/
[Curation Tool v0.1]: https://github.com/phyloref/curation-tool/projects/1
[minimum viable product]: https://en.wikipedia.org/wiki/Minimum_viable_product
[Estimate rate of phyloreference curation]: https://github.com/phyloref/curation-tool/projects/4
[Curation Tool v0.2]: https://github.com/phyloref/curation-tool/projects/2
[Curation Tool v1.0]: https://github.com/phyloref/curation-tool/projects/3
[OWL (Web Ontology Language)]: https://en.wikipedia.org/wiki/Web_Ontology_Language
[Fisher et al., 2006]: http://www.bioone.org/doi/abs/10.1639/0007-2745%282007%29110%5B46%3APOTCWA%5D2.0.CO%3B2
